silkworm/capi/snapshots.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+   Copyright 2025 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#include "snapshots.hpp"
+
+#include <silkworm/db/datastore/snapshots/snapshot_bundle.hpp>
+#include <silkworm/db/state/schema_config.hpp>
+
+using namespace silkworm;
+using namespace silkworm::db;
+using namespace silkworm::snapshots;
+
+MemoryMappedRegion make_region(const SilkwormMemoryMappedFile& mmf) {
+    return {mmf.memory_address, mmf.memory_length};
+}
+
+static std::optional<SnapshotPath> parse_snapshot_path(const char* file_path) {
+    if (!file_path) return std::nullopt;
+    return SnapshotPath::parse(file_path);
+}
+
+SnapshotBundleEntityDataResult build_inverted_index_entity_data(const SilkwormInvertedIndexSnapshot& snapshot) {
+    const auto ii_segment_path = parse_snapshot_path(snapshot.segment.file_path);
+    if (!ii_segment_path) {
+        return tl::make_unexpected(SILKWORM_INVALID_PATH);
+    }
```

Comment:
This pattern repeats too many times and makes code a bit hard to follow.
Since this is a private function, I suggest to a typed exception here thrown from parse_snapshot_path, and add try/catch on the silkworm_add_state_snapshot level, and convert it to a C error there.